src/visualizers/widgets/PlotlyGraph/PlotlyGraphWidget.js

```diff
@@ -64,10 +76,15 @@ define(['./lib/plotly.min',], function (Plotly) {
         this.plotlyJSON = null;
         if (this.created) {
             Plotly.purge(this.$el[0]);
+            this.toggleText();
         }
         this.created = false;
     };
 
+    PlotlyGraphWidget.prototype.toggleText = function (append = true) {
```

Comment:
This is misleading as `toggleText` does not actually toggle the text as shown below:
```javascript
this.toggleText();  // text shown
this.toggleText();  // text hidden
this.toggleText();  // text shown
// etc...
```
The current implementation behaves more like `setTextVisible` or something. However, even in this case, it behaves a little odd as it isn't idempotent (calling it multiple times will result in appending multiple divs rather than just the one).

There are a couple ways to get around this. One approach would be to simply add the div and a class (like `empty`) to the widget and then have a css rule for the `display` property of the text so it only shows up when not in an `empty` class. This method would then be able to be changed to something like `setEmptyTextVisible(isEmpty)` (which just adds or removes the "empty" class).

Reply:
Yeah, that makes sense, I was not sure that `plotly` would keep or remove the child elements of the div on which the graph is drawn (from the DOM). However, it is not the case and child elements are not deleted. So, this function shown below can be used.   
```js
    PlotlyGraphWidget.prototype.setTextVisibility = function (display = true) {
         display = display ? 'block': 'none';
        this.$defaultTextDiv.css('display', display);
    };
```
It could also help to decouple the text visibility calls from graph drawing methods, as they are independent of each other. The above method should be called on `initialize`, `addNode`, `removeNode` and `updateNode` for the widget.

Reply:
Cool. As a minor thing, I probably wouldn't use a default argument for `setTextVisibility` as it isn't obvious that it should be `true` by default. I usually use default values for arguments that are conceptually optional as well (such as "max number of results when querying a database", "initial value to `reduce`", etc). It's just a little awkward to see a call to `setTextVisibility` where the value that we are setting it to is omitted.